resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/features/MountVolumesFeatureStepSuite.scala

```diff
@@ -148,6 +149,23 @@ class MountVolumesFeatureStepSuite extends SparkFunSuite {
     assert(executorPVC.getClaimName.endsWith("-exec-1-pvc-0"))
   }
 
+  test("SPARK-39006 PVC claimName must be onDemand when multiple executors") {
+    val volumeConf = KubernetesVolumeSpec(
+      "testVolume",
+      "/tmp",
+      "",
+      mountReadOnly = true,
+      KubernetesPVCVolumeConf("testClaimName")
+    )
+    val conf = new SparkConf().set(EXECUTOR_INSTANCES, 2)
+    val executorConf =
+      KubernetesTestConf.createExecutorConf(sparkConf = conf, volumes = Seq(volumeConf))
+    val executorStep = new MountVolumesFeatureStep(executorConf)
+    assertThrows[IllegalArgumentException] {
```

Comment:
IMO it would be good to assert on the exception message too. Or use a specialized exception.

Reply:
Thanks for your help. I have addressed this comment.